networkd/network_server.ml

```diff
@@ -336,6 +336,32 @@ module Interface = struct
 			Sysfs.is_physical name
 		) ()
 
+	let has_vlan _ dbg ~name ~vlan =
+		(* Identify the vlan is used by kernel which is unknown to XAPI *)
+		Debug.with_thread_associated dbg (fun () ->
+			let vlan_ifaces =
+				try
+					let vlan_config = String.split '\n' (Unixext.string_of_file "/proc/net/vlan/config") in
+					(* First 2 lines from config file are not needed *)
+					let vlan_config_list =
+						match vlan_config with
+						| a::b::l -> l
+						| _ -> []
+					in
+					(* We need only device and VLAN ID from each line of config file *)
+					let create_pair lst =
+						match (Re_str.split (Re_str.regexp_string "| ") lst) with
```

Comment:
I believe that the `Re_str` module is not thread-safe. We've had issues with it in the past.